### DISCUSSION TITLE
Upgrade from PHP version 7

### DISCUSSION THREAD

**Discussion by MerlinB** [nextcloud/all-in-one#4619] [Questions] [answered]
Hello. After updating, I am stuck with `This version of Nextcloud is not compatible with PHP>=8.2.`. I determined that my installed Nextcloud is only compatible with PHP version 7.x. To resolve this I followed the guide at https://github.com/nextcloud/all-in-one/blob/main/manual-upgrade.md. Unfortunately, there exist no docker images for PHP versions below 8 that I could find to complete the update as described in step 6. Is there a way out of this?

**Comment by szaimen** [comment, marked as answer]
Hi, you should be able to use e.g. the 8.1 tag

**Reply by MerlinB** [reply]
Sorry, I was mistaken, it needs PHP 8.0, not PHP 7.

Still, I did change the container to `nextcloud/aio-nextcloud:php8.0-latest` in `/tmp/nextcloud-aio-nextcloud` and ran it, as explained in the instructions. In the following container logs I still get `App "Nextcloud All-in-One" cannot be installed because it is not compatible with this version of the server.`.

Here is the content of `versioncheck.php`:
```
// Show warning if a PHP version below 7.4 is used,
if (PHP_VERSION_ID < 70400) {
	http_response_code(500);
	echo 'This version of Nextcloud requires at least PHP 7.4<br/>';
	echo 'You are currently running ' . PHP_VERSION . '. Please update your PHP version.';
	exit(1);
}

// Show warning if >= PHP 8.2 is used as Nextcloud is not compatible with >= PHP 8.2 for now
if (PHP_VERSION_ID >= 80200) {
	http_response_code(500);
	echo 'This version of Nextcloud is not compatible with PHP>=8.2.<br/>';
	echo 'You are currently running ' . PHP_VERSION . '.';
	exit(1);
}
```

**Reply by MerlinB** [reply]
I figured out that while `nextcloud/aio-nextcloud:php8.0-latest` and `nextcloud/aio-nextcloud:latest` did not work, `nextcloud/aio-nextcloud:php8.1-latest` did work.
